### PR TITLE
[storage][pruner] Add explicit config to enable/disable pruners

### PR DESCRIPTION
**Pull request by zcchahaha**
### Description
- Add configs to explicitly enable/disable pruners instead of depending on the pruner windows.
- Make pruner window non option.
### Test Plan
Existing Tests.

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2598)

### CODE REVIEW

File: config/src/config/storage_config.rs [hunk 1]
```diff
@@ -88,22 +88,29 @@ pub struct StorageConfig {
...
 };
 
 #[derive(Clone, Copy, Debug, Deserialize, Eq, PartialEq, Serialize)]
 #[serde(deny_unknown_fields)]
 pub struct StoragePrunerConfig {
-    /// None disables pruning. The size of the window should be calculated based on disk space
-    /// availability and system TPS.
-    pub state_store_prune_window: Option<u64>,
+    /// Booleans to enable/disable the state store pruner. The state pruner is responsible for
+    /// pruning SMT states.
```
Comment (msmouse) [inline config/src/config/storage_config.rs:103]: "SMT states" -> "state tree nodes"

File: state-sync/storage-service/server/src/lib.rs [hunk 2]
```diff
@@ -956,29 +956,30 @@ impl StorageReader {
         latest_version: Version,
         transactions_range: &Option<CompleteDataRange<Version>>,
     ) -> Result<Option<CompleteDataRange<Version>>, Error> {
+        let pruner_enabled = self.storage.is_state_pruner_enabled().map_err(|error| Error::StorageErrorEncountered(error.to_string()))?;
```
Comment (msmouse) [inline state-sync/storage-service/server/src/lib.rs:959]: We now know the real range of available state trees (even regardless if the pruner is enabled), shall we just return the real thing? @JoshLind

(and `is_state_pruner_enabled()` isn't useful in the case)

**Review by msmouse — APPROVED** [review]

File: state-sync/storage-service/server/src/lib.rs (hunk 2 above)
Comment (zcchahaha) [reply state-sync/storage-service/server/src/lib.rs]: Discussed offline. We can do it in another PR.

**Review by sherry-x — APPROVED** [review]

File: execution/executor-benchmark/src/main.rs [hunk 3]
```diff
@@ -12,11 +12,17 @@ static ALLOC: jemallocator::Jemalloc = jemallocator::Jemalloc;
...
 struct PrunerOpt {
+    #[structopt(long)]
+    enable_state_store_pruner: bool,
+
+    #[structopt(long)]
+    enable_ledger_pruner: bool,
+
     #[structopt(long, default_value = "100000", help = "Set to -1 to disable.")]
-    state_prune_window: i64,
+    state_prune_window: u64,
```
Comment (grao1991) [inline execution/executor-benchmark/src/main.rs:22]: We should only enable this flag when the corresponding boolean is true.

use "requires", e.g.:
https://github.com/aptos-labs/aptos-core/blob/68215374a17c99cf9f90dff273a309f7f35017ae/crates/transaction-emitter-lib/src/args.rs#L87

Comment (zcchahaha) [reply execution/executor-benchmark/src/main.rs:22]: Done.

**Review by grao1991 — APPROVED** [review]

**Pull request merged**